app/pages/events/event_list.php

```diff
@@ -5,6 +5,8 @@
 formatter("d MMM");
 $user = User::getCurrent();
 $events = Event::listAllOpen($user->id);
+$past_events = Event::listAllPastOpen($user->id);
+$future_events = Event::listAllFutureOpen($user->id);
```

Comment:
This is a lot of overfetching. You could do this instead:
- Solution 1:  keep the normal query as is, and filter only in PHP using collection methods.
- Solution 2: have a lazy-loading boundary with `hx-trigger=load`, since most people will not be interested in the past event anyways.

Solution 2 is more scalable (with a lot of events, it will still perform well) and imo more fun to implement, but solution 1 is a lot easier so you can go with this one if you want!

Reply:
Chose solution 2, more fun with htmx